website/static/js/accountSettings.js

```diff
@@ -205,7 +211,8 @@ var UserProfileViewModel = oop.extend(ChangeMessageMixin, {
                         return;
                     }
                 }
-                this.changeMessage('Invalid Email.', 'text-danger');
+            }.bind(this)).fail(function(){
+                this.profile().emails.remove(email);
```

Comment:
I'm not sure I understand; implementing what Andy is suggesting should be as simple as moving `this.profile().emails.push(email);` to the `done` callback, right?